pandas/tests/reshape/test_util.py

```diff
@@ -4,31 +4,52 @@
 import pandas.util.testing as tm
 from pandas.core.reshape.util import cartesian_product
 
+import string
+from datetime import date
+from dateutil import relativedelta
+
+from pandas.util import _hypothesis as hp
+
+NO_OF_EXAMPLES_PER_TEST_CASE = 20
+
 
 class TestCartesianProduct(object):
 
-    def test_simple(self):
-        x, y = list('ABC'), [1, 22]
+    @hp.settings(max_examples=20)
```

Comment:
so this is a lot of boilerplate here, this has to be simpler, we have thousands of tests (sure most cannot use this), but I think you would need to have this reduced to basically a 1-liner to have people use it. can you put some functions in _hypthesis to make this much more readable.

this comment for each of the additions here.

Reply:
"""so this is a lot of boilerplate here, this has to be simpler, we have thousands of tests (sure most cannot use this), but I think you would need to have this reduced to basically a 1-liner to have people use it. can you put some functions in _hypthesis to make this much more readable."""

If you are talking about boilerplate, like below decorators, I don't think, it can be further reduced as just like, we are bound to have different parametrize for different test cases in pytest, the same issue is with below code. For some function it will be less, for other it can be more. 
may be when code evolves, more common code comes out to be refactored.

@hp.settings(max_examples=20)
    @hp.given(hp.st.lists(hp.st.text(string.ascii_letters, min_size=1, max_size=1),
                          min_size=1, max_size=3),
           hp.get_seq((int,), False, 1, 2))
    def test_simple(self, x, y):